### PR TITLE
Add hook-std typings

### PR DESCRIPTION
**Pull request by Codex-**
Typings & tests created for [hook-std](https://github.com/sindresorhus/hook-std)

@sindresorhus as the author (and seemingly a very busy person) could you please also review these typings also?

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by sindresorhus** [comment]
For what it's worth, I accept TS definitions for my modules if you want to submit it directly to `hook-std`. I even have a style guide to make it easier to submit: https://github.com/sindresorhus/typescript-definition-style-guide

File: types/hook-std/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+export type Unhook = () => void;
+
+/**
+ * Receives stdout/stderr as the first argument and the unhook method as the
+ * second argument. Return a string to modify it.
+ *
+ * Optionally, when in silent mode, you may return a boolean to influence
+ * the return value of .write(...).
+ */
+export type Transform = (output: string, unhook: Unhook) => void;
```
Comment (sindresorhus) [inline types/hook-std/index.d.ts]: > Optionally, when in silent mode, **you may return a boolean** to influence
 * the return value of .write(...).

File: types/hook-std/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+	 */
+	streams?: Writable[];
+}
+
+/**
+ * Promise with a unhook() method which, when called, resolves the Promise
+ * with an empty result.
+ */
+export interface PromiseUnhook extends Promise<any> {
+	unhook: () => void;
```
Comment (sindresorhus) [inline types/hook-std/index.d.ts]: I would use the `Unhook` type here.

Comment (Codex-) [reply types/hook-std/index.d.ts]: Good catch, thanks

File: types/hook-std/index.d.ts (hunk 1 above)
Comment (Codex-) [reply types/hook-std/index.d.ts]: Ah, my mistake.
`export type Transform = (output: string, unhook: Unhook) => void | boolean;`

**Comment by Codex-** [comment]
@sindresorhus if these look correct I can migrate them to a PR with the `hook-std` repository, would be nicer.

**Comment by minestarks** [comment]
@sindresorhus we would love it if you published typings with your package! Holding off on merging this in case you do.

**Comment by Codex-** [comment]
@minestarks I've actually almost finished migrating these to the standard @sindresorhus has specified (just writing tests) and will be making a PR soon on the `hook-std` repo, so will close this one. Thanks for looking over this PR :)

**Pull request closed without merging**